src/js/actions/tools.js

```diff
@@ -36,6 +38,41 @@ define(function (require, exports) {
         policy = require("./policy"),
         shortcuts = require("./shortcuts");
         
+
+    /**
+     * Installs the defaults on a given shape tool
+     * 
+     * @param {string} toolName the name of the tool we're using "ellipseTool" or "rectangleTool"
+     * @param {Color} strokeColor a 3 item array represetning the [r,g,b] value of the stroke
+     * @param {number} strokeWidth the width of the stroke
+     * @param {number} strokeOpacity the opacity of the stroke
+     * @param {Color} fillColor a 3 item array represetning the [r,g,b] value of the fill
+     *
+     * @return {Promise}
+     */
+    var installShapeDefaultsCommand = function (tool, strokeColor, strokeSize, strokeOpactiy, fillColor) {
+        var document = this.flux.store("application").getCurrentDocument();
+
+        var defaultObj = toolLib.defaultShapeTool(tool, strokeColor, strokeSize, strokeOpactiy, fillColor);
+        // If document doesn't exist, or is a flat document
+        if (!document || document.unsupported || document.layers.all.size === 1 &&
+            document.layers.all.first().isBackground) {
+            return descriptor.PlayObject(toolLib.resetShapeTool());
+        }
+        var layerSpec = document.layers.allSelected.toList();
```

Comment:
nit: if you're going to short-circuit above, then might as well move `defaultObj` dow to here.  In general, this function seems to to be heavy on individual `var` statements, which differs from our usual pattern.